jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/actions/OpenStreamInEditor.kt

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.actions
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileTypes.PlainTextLanguage
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiFileFactory
+import kotlinx.coroutines.withContext
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.resources.message
+import kotlin.coroutines.ContinuationInterceptor
+
+object OpenStreamInEditor {
+    suspend fun open(project: Project, edt: ContinuationInterceptor, logStream: String, fileContent: String) {
+        val factory = PsiFileFactory.getInstance(project)
```

Comment:
You can use LightVirtualFile directly instead of going through the PSI factory

Reply:
I'm not sure what that saves and that doesn't give the option for `noSizeLimit`

Reply:
LightVirtualFile just takes a CharSequence so it has no size limit already

Reply:
Changed to `ReadOnlyLightVirtualFile`